Sources/Task.swift

```diff
@@ -508,10 +508,22 @@ extension Task {
 						}
 						
 						observer.send(value: .launch(self))
-						process.launch()
-						close(stdoutPipe.writeFD)
-						close(stderrPipe.writeFD)
 
+						do {
+							defer {
+								close(stdoutPipe.writeFD)
+								close(stderrPipe.writeFD)
+							}
+
+							if #available(macOS 10.13, *) {
+								try process.run()
+							} else {
+								process.launch()
+							}
+						} catch {
```

Comment:
Can you move this `do`/`catch` inside the `if` so that it wraps just the `try` statement?

IMO that makes the code clearer.

Reply:
changed it